shared/README.md

```diff
@@ -136,26 +137,119 @@ You'll get various prompts about installing various tools. You should
+
+# if this fails with something like "xcrun: error: unable to lookup item
+# 'Path' in SDK 'iphoneos'" you might have a partial xcode install. try
 xcode-select --switch /Applications/Xcode.app/Contents/Developer
-```
 
-```sh
 # Start the react native publisher (unlike on iOS, have to do this manually).
-yarn run rn-start
+yarn rn-start android
 ```
 
 Then do "Build > Make Project" and then "Run > Run 'app'".
 
+#### Dealing with Android Studio
+
+Sometimes Android Studio gets into a bad state, especially if you're
+doing stuff like `yarn modules` in the background.
+
+If 'Gradle sync' fails, you can retry it from Tools > Android > 'Sync
+Project with Gradle Files'.
+
+Sometimes, especially after opening Android Studio after a run of
+`yarn modules`, you'll get an "Unsupported Modules Detected" message
+for "react-native-fetch-blob", "react-native-contacts", and
+"react-native-image-picker". This seems to be harmless.
+
+Sometimes you'll also get an "An IDE Error has occured" message. That
+also seems to be harmless, although you may want to resync/reopen the
+project in that case.
+
+If nothing above works, you can try closing (File > Close Project) and
+reopening the project. Or even closing and reopening the app.
+
+#### Building and developing without Android Studio
+
+Alternatively, you can build and develop without Android
+Studio. However, first make sure you _can_ build and run with Android
+Studio first, as it's easier to get that working first.
+
+So make sure you've run `yarn rn-gobuild-android` and you have the
+react-native packager running (`yarn rn-start android`).
+
+Then run
+
+```sh
+# Build the apk.
+yarn rn-build-android
+```
+
+Unless you're modifying the Java files, you likely have to only run
+this occasionally.
+
+Then make sure you either have an emulator running, or you have your
+Android device connected (but not both). To check, run
+
+```sh
+adb devices
+```
+
+It should list exactly one device.
+
+To run the emulator, do:
+
+```sh
+cd $ANDROID_HOME/emulator
+emulator -list-avds
+# Nexus_5X_API_27_x86 is an example avd.
+emulator @Nexus_5X_API_27_x86
+```
+
+To run on your Android device, make sure USB debugging is enabled; see
+[these
+instructions](https://facebook.github.io/react-native/docs/running-on-device.html). Then
+plug in your device via USB and tap 'OK' on the 'Allow USB debugging?'
+prompt if it appears. After that, `adb devices` should list your
+device. If it says 'unauthorized' next to your device, then you likely
+haven't tapped 'OK' on the prompt yet.
+
+Finally, you'll have to forward port 8081 on your device to port 8081
+on your computer. To do so, run
+
+```sh
+adb reverse tcp:8081 tcp:8081
+```
+
+To recap, you should have run `rn-gobuild-android`,
+`rn-build-android`, have the react-native packager running, and either
+have an emulator running, or an Android device connected. `adb
+devices` should list exactly one connected device.
+
+Then you can run
+
+```sh
+yarn rn-push-android
+```
+
```

Comment:
If I've modified Go files, I'd need to re-run `run-gobuild-android`. In this case, do I need to re-run `rn-build-android`, or is the old build gonna automatically use the new Go part?

Reply:
Pretty sure you'd need to rerun both. I'll add a comment.

Reply:
Cool; thanks!